friendshipper/src-tauri/src/repo/operations/reset.rs

```diff
@@ -1,12 +1,59 @@
-use crate::engine::EngineProvider;
-use crate::state::AppState;
-use axum::extract::State;
-use ethos_core::types::errors::CoreError;
-
-pub async fn reset_repo<T>(State(state): State<AppState<T>>) -> Result<(), CoreError>
-where
-    T: EngineProvider,
-{
-    let branch = state.repo_config.read().trunk_branch.clone();
-    state.git().hard_reset(&branch).await.map_err(|e| e.into())
-}
+use crate::engine::EngineProvider;
+use crate::state::AppState;
+use async_trait::async_trait;
+use axum::extract::{Path, State};
+use axum::Json;
+use ethos_core::clients::git;
+use ethos_core::types::errors::CoreError;
+use ethos_core::worker::{Task, TaskSequence};
+
+pub async fn reset_repo<T>(State(state): State<AppState<T>>) -> Result<(), CoreError>
+where
+    T: EngineProvider,
+{
+    let branch = state.repo_config.read().trunk_branch.clone();
+    state.git().hard_reset(&branch).await.map_err(|e| e.into())
+}
+
+#[derive(Clone)]
+pub struct ResetToCommitOp {
+    pub commit: String,
+    pub git_client: git::Git,
+}
+
+#[async_trait]
+impl Task for ResetToCommitOp {
+    async fn execute(&self) -> Result<(), CoreError> {
+        self.git_client
+            .run(&["reset", "--hard", &self.commit], Default::default())
+            .await
+            .map_err(CoreError::Internal)
+    }
```

Comment:
This will also need to run a DownloadDllsOp and UpdateEngineOp as well, since the game and engine code may have changed between commits.

Reply:
I think you'll also need to run a StatusOp before running those downloads as I believe it's the StatusOp that sets the right commit for the DownloadDllsOp to use. I may be wrong.

Reply:
@rudoi in the refresh() call we invoke get_repo_status -> status_handler (StatusOp)

Reply:
i think the reason we're specifically calling out the Ops is so that they're included in the TaskSequence that executes the reset. in its current state, the reset endpoint leaves friendshipper in a little bit of an unknown state, whereas you can use the TaskSequence to ensure that everything you need gets executed. if you do this sequencing in the frontend, there's no guarantee some other queued request won't make some kind of destructive change to the repo. does that make sense?